services/src/api/model/datatypes.rs

```diff
@@ -1032,3 +1296,267 @@ impl TryFrom<SerializablePalette> for Palette {
         Ok(Self(inner))
     }
 }
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Hash, Eq, PartialOrd, Ord, ToSchema)]
+pub struct RasterPropertiesKey {
+    pub domain: Option<String>,
+    pub key: String,
+}
+
+impl From<geoengine_datatypes::raster::RasterPropertiesKey> for RasterPropertiesKey {
+    fn from(value: geoengine_datatypes::raster::RasterPropertiesKey) -> Self {
+        Self {
+            domain: value.domain,
+            key: value.key,
+        }
+    }
+}
+
+impl From<RasterPropertiesKey> for geoengine_datatypes::raster::RasterPropertiesKey {
+    fn from(value: RasterPropertiesKey) -> Self {
+        Self {
+            domain: value.domain,
+            key: value.key,
+        }
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, ToSchema)]
+pub enum RasterPropertiesEntryType {
+    Number,
+    String,
+}
+
+impl From<geoengine_datatypes::raster::RasterPropertiesEntryType> for RasterPropertiesEntryType {
+    fn from(value: geoengine_datatypes::raster::RasterPropertiesEntryType) -> Self {
+        match value {
+            geoengine_datatypes::raster::RasterPropertiesEntryType::Number => Self::Number,
+            geoengine_datatypes::raster::RasterPropertiesEntryType::String => Self::String,
+        }
+    }
+}
+
+impl From<RasterPropertiesEntryType> for geoengine_datatypes::raster::RasterPropertiesEntryType {
+    fn from(value: RasterPropertiesEntryType) -> Self {
+        match value {
+            RasterPropertiesEntryType::Number => Self::Number,
+            RasterPropertiesEntryType::String => Self::String,
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, PartialEq, Eq, Clone, Debug, ToSchema)]
+pub struct DateTimeParseFormat {
+    fmt: String,
+    has_tz: bool,
+    has_time: bool,
+}
+
+impl From<geoengine_datatypes::primitives::DateTimeParseFormat> for DateTimeParseFormat {
+    fn from(value: geoengine_datatypes::primitives::DateTimeParseFormat) -> Self {
+        Self {
+            fmt: value._to_parse_format().to_string(),
+            has_tz: value.has_tz(),
+            has_time: value.has_time(),
+        }
+    }
+}
+
+impl From<DateTimeParseFormat> for geoengine_datatypes::primitives::DateTimeParseFormat {
+    fn from(value: DateTimeParseFormat) -> Self {
+        Self::custom(value.fmt)
+    }
+}
+
+impl DateTimeParseFormat {
+    pub fn custom(fmt: String) -> Self {
```

Comment:
can you use the method from the dataypes crate and the do `into`?

Reply:
Do you mean like this? [44637dd](https://github.com/geo-engine/geoengine/pull/662/commits/44637ddbac75b8e5ddfd9ab6d09f7776eb6cbdab)
`DateTimeParseFormat::unix` is used in `OgrSourceTimeFormat` as default value so I implemented it in api::model.